mrtt-ui/src/components/SiteBackgroundForm.js

```diff
@@ -0,0 +1,383 @@
+    govermentArrangement: Yup.array().of(Yup.string()),
+    landTenure: Yup.array().of(Yup.string()),
+    customaryRights: Yup.string()
+  })
+  const formOptions = { resolver: yupResolver(validationSchema) }
+
+  // get functions to build form with useForm() and useFieldArray() hooks
+  const { handleSubmit, formState, control } = useForm(formOptions)
+  const { errors } = formState
+  const {
+    fields: stakeholdersFields,
+    append: stakeholdersAppend,
+    remove: stakeholdersRemove
+  } = useFieldArray({ name: 'stakeholders', control })
+
+  // state variables
+  const [isSubmitting, setisSubmitting] = useState(false)
+  const [isError, setIsError] = useState(false)
+
+  const onSubmit = async (data) => {
+    setisSubmitting(true)
+    setIsError(false)
+    const preppedData = []
+    // TODO: update url to match form section
+    const url = `${process.env.REACT_APP_API_URL}sites/1/registration_answers`
+
+    if (!data) return
+
+    console.log('data:', data)
+
+    // TODO: make more general & reusable function
+    // set up data structure for api
+    for (const [key, value] of Object.entries(data)) {
+      // map question ids with keys
+      if (Object.prototype.hasOwnProperty.call(questionMapping.projectDetails, key)) {
+        preppedData.push({ question_id: questionMapping.projectDetails[key], answer_value: value })
+      }
+    }
+    // make axios PUT request
+    axios
+      .put(url, preppedData)
+      .then((res) => {
+        setisSubmitting(false)
+        console.log(res)
+      })
+      .catch((error) => {
+        setIsError(true)
+        setisSubmitting(false)
+        console.log(error)
+      })
+  }
+
+  const handleStakeholdersOnChange = (e, stakeholder) => {
+    if (e.target.checked) {
+      stakeholdersAppend({ stakeholderType: stakeholder })
+    } else {
+      const index = stakeholdersFields.findIndex((field) => field.stakeholderType === stakeholder)
+      stakeholdersRemove(index)
+    }
+  }
+
+  // TODO: handle '2.5 & 2.8 other option textfield add in'
+
+  return (
+    <MainFormDiv>
+      {/* Select Stakeholders */}
+      <Typography variant='h4' sx={{ marginBottom: '0.5em' }}>
+        Site Background Form
+      </Typography>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <FormQuestionDiv>
+          <FormLabel>2.1 Which stakeholders are involved in the project activities?</FormLabel>
+          <List>
+            {stakeholderOptions.map((stakeholder, index) => (
+              <ListItem key={index}>
+                <Box>
+                  <Box>
+                    <Checkbox
+                      value={stakeholder}
+                      onChange={(e) => handleStakeholdersOnChange(e, stakeholder)}></Checkbox>
+                    <Typography variant='subtitle'>{stakeholder}</Typography>
+                  </Box>
+                  <Box>
+                    {stakeholdersFields.find((field) => field.stakeholderType === stakeholder) && (
```

Comment:
Whats this function do? Can it have a name?

Reply:
can we bust it out into its own function called `getStakeholder` or `getIsStakeholder` to make the jsx more quickly readable?

Reply:
sure!